code/obj/item/grenades.dm

```diff
@@ -934,8 +935,15 @@ ADMIN_INTERACT_PROCS(/obj/item/gimmickbomb, proc/arm, proc/detonate)
 	proc/beep(i)
 		var/k = i/2
 		sleep(k*k)
-		flick(icon_state+"_beep", src)
+
+		if (icon_state_armed)
+			if (icon_state == src.icon_state)
+				icon_state = icon_state_armed
+			else
+				icon_state = src.icon_state
+
```

Comment:
```suggestion
```

Reply:
add to arm()

```suggestion
		if (icon_state_armed)
			icon_state = src.icon_state
```

Reply:
> add to arm()
> 
> ```
> 	if (icon_state_armed)
> 		icon_state = icon_state_armed
> ```

Done, it actually looks a lot better now since it immediately switches to it's armed animation instead of waiting for the beep, which makes it look more natural and realistic for the pin/lever to immediately go up.